x-pack/plugins/security/server/authentication/authentication_result.test.ts

```diff
@@ -296,6 +296,14 @@ describe('AuthenticationResult', () => {
       expect(authenticationResult.userProfileGrant).toBe(userProfileGrant);
       expect(authenticationResult.error).toBeUndefined();
     });
+
+    it('should be able to update the redirectURL after the AuthenticationResult has been created', () => {
+      const newRedirectURL = '/some/other/url';
+      const authenticationResult = AuthenticationResult.redirectTo('/redirect/url');
+      authenticationResult.redirectURL = newRedirectURL;
+
+      expect(authenticationResult.redirectURL).toBe(newRedirectURL);
```

Comment:
We originally made all the properties of `AuthenticationResult` read-only once the object was created. This was primarily done as a caution (@azasypkin can probably fill in more historical details). If we have good use-cases of wanting to edit them post-creation, we can decide to make them editable on a case by case basis. So that's what I'm doing in this PR.

Reply:
An alternative approach which wouldn't require making the `redirectUrl` editable would be the one I just described in this comment: https://github.com/elastic/kibana/pull/139811#discussion_r1001373123